### PR TITLE
Fix an error when dragging nodes into built-in scripts because script does not inherit Node

### LINKED ISSUE #81294: Dragging nodes into built-in scripts causes error: "Can't drop nodes because script 'xxxx::xxxxxxx' does not inherit Node"
### Godot version

v4.2.dev3.mono.official [013e8e3af]

### System information

Godot v4.2.dev3.mono - Windows 10.0.22621 - Vulkan (Forward+) - dedicated NVIDIA GeForce RTX 3050 Ti Laptop GPU (NVIDIA; 31.0.15.3598) - AMD Ryzen 7 5800H with Radeon Graphics (16 Threads)

### Issue description

When creating and attaching a built-in script to any type of node, trying to drag nodes from the scene-tree results in an error that appears at the bottom-right of the editor, but not in the console window. The error states:

`Can't drop nodes because script 'xxxxxxxx::xxxxxxx' does not inherit Node.`
 Even though the script clearly extends Node or any other type.

### Steps to reproduce

1. Create and add any node type to the scene tree.
2. Attach a new script, choose Built-in from the dialog.
3. Save the scene first, as at the time of writing, Godot suffers from a bug that causes it to crash if the scene was not saved first.
4. Try dragging or even ctrl-dragging a node to the script.
5. Observe the error appear at the bottom-right in a yellow-accent pop-up.
6. Nod and confirm that the bug exists.

### Minimal reproduction project

[can't drag nodes into built-in scripts.zip](https://github.com/godotengine/godot/files/12511196/can.t.drag.nodes.into.built-in.scripts.zip)

Loose files in the archive. Extract into their own folder.

### PR DESCRIPTION
**Pull request by jsjtxietian**
Fixes #81294 
Reload built_in script after create to allow drag nodes into it. Calling `reload` can compile the script and give it the inheritance info, thus can pass the check in `drop_data_fw`.

### CODE REVIEW

**Comment by akien-mga** [comment]
> Reload built_in script after create to allow drag nodes into it. Calling `reload` can compile the script and give it the inheritance info, thus can pass the check in `drop_data_fw`.

This probably warrants a comment in the code, as it's not obvious why a script that has just been created would need to be reloaded. I also wonder if it's the right location for it or if it should be done after receiving `script_created`?

**Comment by jsjtxietian** [comment]
> I also wonder if it's the right location for it or if it should be done after receiving `script_created`

As far as I can see, for no built_in scripts, it will get reloaded and compiled in `Error err = ResourceSaver::save(scr, lpath, ResourceSaver::FLAG_CHANGE_PATH)`, rightly after it is created. But there might be other reasons why we can't do the same for built_in scripts that I do not know.

File: editor/script_create_dialog.cpp [hunk 1]
```diff
@@ -390,6 +390,8 @@ void ScriptCreateDialog::_create_new() {
 
 	if (is_built_in) {
 		scr->set_name(internal_name->get_text());
+		// Make sure the script template is compiled thus allow drop nodes into the newly created built_in script.
```
Comment (AThousandShips) [inline editor/script_create_dialog.cpp:393]:
```suggestion
		// Make sure the script template is compiled, allowing dropping nodes into the newly created built-in script.
```

**Review by KoBeWi — APPROVED** [review]
The comment might need rewording (not sure about template; also instead of specifically referring to dropping nodes I think it should say that it makes type recognizable or something), but I can confirm this change fixes the issue.

I checked when external script are reloaded and it happens on save as you noted, but only when `auto_reload_and_parse_scripts_on_save` editor setting is enabled. Otherwise the script will be reloaded when generating preview 🙃 (which also happens right after saving)

So I think this is a correct place to reload the script. Although it's weird that it isn't done automatically somewhere (in script editor maybe).

**Comment by jsjtxietian** [comment]
> The comment might need rewording.

I will update it soon.

**Review by YuriSizov — APPROVED** [review]

**Comment by YuriSizov** [comment]
Thanks!

**Pull request merged**